src/library.js

```diff
@@ -1714,25 +1714,25 @@ LibraryManager.library = {
   //    being compiled. Not sure how to tell LLVM to not do so.
   // ==========================================================================
 
+#if MAIN_MODULE == 0
+  dlopen: function(/* ... */) {
+    abort("To use dlopen, you need to use Emscripten's linking support, see https://github.com/kripken/emscripten/wiki/Linking");
+  },
+  dlclose: 'dlopen',
+  dlsym:   'dlopen',
+  dlerror: 'dlopen',
+  dladdr:  'dlopen',
+#else
+
   $DLFCN: {
```

Comment:
We should either have both `DLFCN` and `LDSO` objects in the library here, or both in support.js there. Unless there is a reason to keep them separate. Would LDSO be used without dlopen?

Reply:
Yes, LDSO (`ld.so` stands for dynamic linker/loader, see e.g. http://man7.org/linux/man-pages/man8/ld.so.8.html) is always used in `loadDynamicLibrary` who anyway needs it to load dynamic libraries that were linked at compile time to the program (or to another library after my next patch).

DLFCN is used only by dlopen & friends, which are in `-ldl` (libdl), delegates most of the work to the dynamic linker, but still needs to store some libdl specific bits somewhere (e.g. error to return from `dlerror`) - thus DLFCN.

This way LDSO should be alway present (e.g. in support.js) and DLFCN should be present only when libdl is in use (thus in library.js).